### PR TITLE
Andreea g/tema5

### PR DESCRIPTION
**Pull request by redandreea**

### CODE REVIEW

File: src/main/java/AndreeaG/driver/BrowserManager.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package AndreeaG.driver;
+
+public class BrowserManager {
```
Comment (95teo) [inline src/main/java/AndreeaG/driver/BrowserManager.java:3]: You are missing in this class the method that creates the driver. Please add it and use it in your methods. Thanks!

File: src/test/java/AndreeaG/teorie/SeleniumBasicsPart2/ActionsExample.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package AndreeaG.teorie.SeleniumBasicsPart2;
```
Comment (95teo) [inline src/test/java/AndreeaG/teorie/SeleniumBasicsPart2/ActionsExample.java:1]: All these theory classes shouldn't be in this PR. Please remove them. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import teofilursan.driver.BrowserManager;
+
+public class Buttons {
+
+
+    public static void main(String[] args) {
+     clickButtons();
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:13]: No need to leave one blank line before the closing '}' at the end of methods. You can remove that line. Check on other methods from all the classes too. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        driver.get("https://demoqa.com/buttons");
+        WebElement doubleClickMe = driver.findElement(By.id("doubleClickBtn"));
+        doubleClickMe.click();
+        System.out.println("Clicked on Double Click me button and text shown is: " +doubleClickMe.getText());
+
+        WebElement rightClickMe = driver.findElement(By.id("rightClickBtn"));
+        rightClickMe.click();
+        System.out.println("Clicked on Right Click me button the text shows: " +rightClickMe.getText());
+
+        WebElement clickMeClickMe = driver.findElement(By.xpath("/html/body/div[2]/div/div/div[2]/div[2]/div[2]/div[3]/button"));
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:27]: Replace the absolute xpaths with relative xpaths. Applicable for all classes that use absolute xpaths. You can find some example in other PRs too. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/CheckBox.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    public static void goToCheckbox1() {
+
+        ChromeDriver driver = BrowserManager.createChromeDriver();
+        driver.get("https://demoqa.com/checkbox");
+
+        WebElement checkBox = driver.findElement(By.xpath("/html/body/div[2]/div/div/div[2]/div[2]/div[2]/div/ol/li/span/label/span[1]"));
+        checkBox.click();
+
+
+        WebElement expandPlusSign =driver.findElement(By.cssSelector("html > body > div:nth-of-type(2) > div > div > div:nth-of-type(2) > div:nth-of-type(2) > div:nth-of-type(2) > div:first-of-type > div > button:first-of-type"));
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/CheckBox.java:24]: Try to shorten this cssSelector too as it's starting straight form the html root tag. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java (hunk 3 above)
Comment (redandreea) [reply src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:13]: I will clean them all up! thank you!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/CheckBox.java (hunk 5 above)
Comment (redandreea) [reply src/test/java/AndreeaG/tests/tema5/Demoqa/CheckBox.java:24]: re-ran with By.cssSelector(".rct-icon.rct-icon-expand-all"

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java (hunk 4 above)
Comment (redandreea) [reply src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:27]: I used : //div[3]/button instead.

ty!

Comment (redandreea) [reply src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:27]: replaced all absolute xpaths (I think :)

File: src/test/java/AndreeaG/tests/tema5/Heroku/HTMLTableTag.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import teofilursan.driver.BrowserManager;
+
+import java.util.List;
+
+
+public class HTMLTableTag {
+    public static void main(String[] args) {
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/HTMLTableTag.java:13]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Heroku/ElementAttributesExamples.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package AndreeaG.tests.tema5.Heroku;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import teofilursan.driver.BrowserManager;
+
+public class ElementAttributesExamples {
+    public static void main(String[] args) {
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/ElementAttributesExamples.java:10]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+public class BasicWebPageExample {
+    public static void main(String[] args) {
+        landOnBasicWebPageExample();
+    }
+
+        public static void landOnBasicWebPageExample() {
+
+            ChromeDriver driver = BrowserManager.createChromeDriver();
+            driver.get ("https://testpages.herokuapp.com/styled/index.html");
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java:17]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+            WebElement getFirstPara = driver.findElement(By.className("explanation"));
+            System.out.println("First paragraph:  " + getFirstPara.getText());
+
+           WebElement getMiddlePara = driver.findElement(By.id("para1"));
+           System.out.println("Middle paragraph:  " + getMiddlePara.getText());
+
+
+           WebElement getLastPara = driver.findElement(By.cssSelector("p#para2.sub"));
+           System.out.println("First paragraph:  " + getLastPara.getText());
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java:32]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+           WebElement firstLink = driver.findElement(By.id("basicpagetest"));
+           System.out.println("First link displays:  " + firstLink.getText());
+           firstLink.click();
+
+            WebElement getFirstPara = driver.findElement(By.className("explanation"));
+            System.out.println("First paragraph:  " + getFirstPara.getText());
+
+           WebElement getMiddlePara = driver.findElement(By.id("para1"));
+           System.out.println("Middle paragraph:  " + getMiddlePara.getText());
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/BasicWebPageExample.java:28]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Heroku/HTMLTableTag.java [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+
+
+
+        ChromeDriver driver = BrowserManager.createChromeDriver();
+        driver.get ("https://testpages.herokuapp.com/styled/tag/table.html");
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Heroku/HTMLTableTag.java:19]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Links.java [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import teofilursan.driver.BrowserManager;
+
+public class Links {
+
+    public static void main(String[] args) {
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/Links.java:11]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Links.java [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+
+        ChromeDriver driver = BrowserManager.createChromeDriver();
+        driver.get ("https://demoqa.com/links");
+
+        WebElement homeLink = driver.findElement(By.id("simpleLink"));
+        System.out.println("Home link displays:  " + homeLink.getText());
+        homeLink.click();
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/Links.java:19]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/RadioButton.java [hunk 14]
```diff
@@ -0,0 +1,27 @@
...
+import teofilursan.driver.BrowserManager;
+
+public class RadioButton {
+
+    public static void main(String[] args) {
+        radioButton();
+    }
+
+    public static void radioButton(){
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/RadioButton.java:15]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java [hunk 15]
```diff
@@ -0,0 +1,48 @@
...
+import teofilursan.driver.BrowserManager;
+
+public class WebTables {
+
+    public static void main(String[] args) {
+        accessWebTables();
+    }
+
+    public static void accessWebTables() {
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java:15]: Please remove these unnecessary blank lines from here. Thanks!

File: src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java [hunk 16]
```diff
@@ -0,0 +1,48 @@
...
+
+        ChromeDriver driver = BrowserManager.createChromeDriver();
+        driver.get("https://demoqa.com/webtables");
+        WebElement addNewRecordButton = driver.findElement(By.id("addNewRecordButton"));
+        addNewRecordButton.click();
+
+        WebElement firstName = driver.findElement(By.id("firstName"));
+        firstName.sendKeys( "Ana");
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java:24]: One line is enough.

File: src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java [hunk 17]
```diff
@@ -0,0 +1,48 @@
...
+
+        WebElement lastName = driver.findElement(By.id("lastName"));
+        lastName.sendKeys("Anescu");
+
+        WebElement emailInput = driver.findElement(By.id("userEmail"));
+        emailInput.sendKeys("a.anescu@getbetter.com");
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java:31]: One line is enough.

File: src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java [hunk 18]
```diff
@@ -0,0 +1,48 @@
...
+        WebElement ageNumber = driver.findElement(By.id("age"));
+        ageNumber.sendKeys("99");
+
+        WebElement salaryAmount = driver.findElement(By.id("salary"));
+        salaryAmount.sendKeys("15000");
+
+        WebElement departmentName = driver.findElement(By.id("department"));
+        departmentName.sendKeys(" Quality");
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/WebTables.java:41]: One line is enough.

File: src/test/java/AndreeaG/tests/tema5/Demoqa/TextBox.java [hunk 19]
```diff
@@ -0,0 +1,64 @@
...
+
+        goToTextBox();
+        goToEmail();
+        goToCurrentAddress();
+        goToPermanentAddress();
+        submitButton();
+    }
+
+    private static void goToTextBox() {
+
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/TextBox.java:20]: Please remove these unnecessary blank lines from here. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by ioanaandroflorea — APPROVED** [review]

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java [hunk 20]
```diff
@@ -0,0 +1,29 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import teofilursan.driver.BrowserManager;
+
+public class Buttons {
+
+    public static void main(String[] args) {
+     clickButtons();
+    }
+    public static void clickButtons(){
```
Comment (95teo) [inline src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:13]: It's ok to leave one blank line between methods. So please make also this change in all classes. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java (hunk 20 above)
Comment (redandreea) [reply src/test/java/AndreeaG/tests/tema5/Demoqa/Buttons.java:14]: Fingers crossed :)

**Review by tincageorgianaalina — APPROVED** [review]

**Review by 95teo — APPROVED** [review]

**Pull request merged**